metadata-models/src/main/pegasus/com/linkedin/common/ContainerPathEntry.pdl

```diff
@@ -0,0 +1,17 @@
+namespace com.linkedin.common
+
+/**
+ * Represents a single level in an entity's container path
+ */
+record ContainerPathEntry {
+  /**
+   * The ID of the container path entry - what gets stored in the index, URL encoded
+   * If there's an urn associated with this entry, id and urn will be the same
+   */
+  id: string
```

Comment:
Could we ever want to use container paths to support a more custom "folder" type experience, where users can choose what path they want entities in, especially those that don't fit in containers? I think the lines are a little blurred because ContainerPaths aren't just for storing the path of containers for an entity, it's the path of containers for an entity + another path for entities that aren't stored in containers.

I know this is a significant change, but what do you think about making container paths be composed of _only_ containers (thus can be a list of urns), then having a second aspect like `CustomBrowsePath` that we use for the non-container paths?

Reply:
hmm yeah i definitely get what you're saying. Thinking about implementation, I personally would rather not have to know whether a given entity has `containerPaths` or some other aspect in order for us to operate browse.

The other thing is that we're going to need to filter on this aspect across all entity types (when they click a group on the sidebar of browse) so having that custom logic to filter for one field on elastic or another may be frustrating.

Maybe we can come up with a better name other than `containerPaths`? however the word "container" is already generic enough - it's just a little overloaded since we have actual "container" entities.

what do you think?

Reply:
I think the name is ok, maybe there's a better one but I don't feel too strongly about it. Thinking over it more, this seems reasonable enough to me!